internal/net/httpserver.go

```diff
@@ -36,12 +42,15 @@ type HTTPServer struct {
 	lock     sync.Mutex
 	listener net.Listener
 	done     chan error
-	stopped  uint32
+	stopped  atomic.Bool
```

Comment:
We're reading and writing the atomic before acquiring the lock.

Sure it's not a hot path and we could get away with a plain bool and the
existing lock. I can change it if you feel strongly about that.